src/runtime/inspector_client.hpp

```diff
@@ -40,12 +46,24 @@ class TrInspectorClient
   void respond(http::Response response);
   void respond(uint32_t code, const std::string &text);
   void respond(uint32_t code, const rapidjson::Document &json);
+  bool isWebSocket() const
+  {
+    return connectionType_ == ConnectionType::WEBSOCKET;
+  }
+  void sendWebSocketMessage(const string &message);
 
 private:
   bool setNonBlocking();
   void recv();
-  void send(const std::string &data);
+  void send(const string &data);
   void end();
+  void sendHttpErrorResponse(uint32_t code, const string &message);
+
+  // WebSocket related methods
+  bool tryUpgradeToWebSocket();
+  void handleWebSocketFrame();
+  void sendWebSocketFrame(const string &data);
+  string generateWebSocketAcceptKey(const string &webSocketKey);
```

Comment:
```suggestion
  std::string generateWebSocketAcceptKey(const std::string &webSocketKey);
```
In header, use complete signature.